380.py

```diff
@@ -0,0 +1,52 @@
+class RandomizedSet:
+
+    def __init__(self):
+        """
+        Initialize your data structure here.
+        """
+        self.dict = dict()
+        self.list = []
+
+
+    def insert(self, val: int) -> bool:
+        """
+        Inserts a value to the set. Returns true if the set did not already contain the specified element.
+        """
+        if val in self.dict:
+            return False
+
+        self.dict[val] = len(self.list)
+        self.list.append(val)
+
+        return True
+
+
+    def remove(self, val: int) -> bool:
```

Comment:
LGTM. `insert()` and `getRandom()` are trivial; the most difficult part of this problem is to come up with this "remove the last and overwrite something in the middle" trick for `remove()`.